compiler/one-cmds/one-build

```diff
@@ -0,0 +1,536 @@
+
+OPTS=$(getopt --options "$SHORT_OPTS" --longoptions "$LONG_OPTS" --name "$0" -- "$@")
+
+if [ $? != 0 ] ; then echo "[ERROR] Failed to parse options" ; exit 2 ; fi
+
+eval set -- "$OPTS"
+
+unset CONFIG_FILE
+
+while true ; do
+  case "$1" in
+    -h|--help)
+      Usage
+      exit 0
+      ;;
+    -v|--version)
+      Version
+      exit 0
+      ;;
+    -c|--config)
+      CONFIG_FILE="$2"
+      shift 2
+      ;;
+    --input_path)
+      INPUT_PATH="$2"
+      shift 2
+      ;;
+    --output_path)
+      OUTPUT_PATH="$2"
+      shift 2
+      ;;
+    --use_one_import_tf)
+      USE_ONE_IMPORT_TF="$2"
+      shift 2
+      ;;
+    --use_one_import_bcq)
+      USE_ONE_IMPORT_BCQ="$2"
+      shift 2
+      ;;
+    --import_input_arrays)
+      IMPORT_INPUT_ARRAYS="$2"
+      shift 2
+      ;;
+    --import_input_shapes)
+      IMPORT_INPUT_SHAPES="$2"
+      shift 2
+      ;;
+    --import_output_arrays)
+      IMPORT_OUTPUT_ARRAYS="$2"
+      shift 2
+      ;;
+    --import_v2)
+      IMPORT_V2="$2"
+      shift 2
+      ;;
+    --import_tf_output_path)
+      IMPORT_TF_OUTPUT_PATH="$2"
+      shift 2
+      ;;
+    --import_bcq_output_path)
+      IMPORT_BCQ_OUTPUT_PATH="$2"
+      shift 2
+      ;;
+    --import_tflite_output_path)
+      IMPORT_TFLITE_OUTPUT_PATH="$2"
+      shift 2
+      ;;
+    --optimize_all)
+      OPTIMIZE_ALL="$2"
+      shift 2
+      ;;
+    --optimize_fuse_bcq)
+      OPTIMIZE_FUSE_BCQ="$2"
+      shift 2
+      ;;
+    --optimize_fuse_instnorm)
+      OPTIMIZE_FUSE_INSTNORM="$2"
+      shift 2
+      ;;
+    --optimize_resolve_customop_add)
+      OPTIMIZE_RESOLVE_CUSTOMOP_ADD="$2"
+      shift 2
+      ;;
+    --optimize_resolve_customop_batchmatmul)
+      OPTIMIZE_RESOLVE_CUSTOMOP_BATCHMATMUL="$2"
+      shift 2
+      ;;
+    --optimize_resolve_customop_matmul)
+      OPTIMIZE_RESOLVE_CUSTOMOP_MATMUL="$2"
+      shift 2
+      ;;
+    --optimize_output_path)
+      OPTIMIZE_OUTPUT_PATH="$2"
+      shift 2
+      ;;
+    --quantize_input_dtype)
+      QUANTIZE_INPUT_DTYPE="$2"
+      shift 2
+      ;;
+    --quantize_quantized_dtype)
+      QUANTIZE_QUANTIZED_DTYPE="$2"
+      shift 2
+      ;;
+    --quantize_granularity)
+      QUANTIZE_GRANULARITY="$2"
+      shift 2
+      ;;
+    --quantize_min_percentile)
+      QUANTIZE_MIN_PERCENTILE="$2"
+      shift 2
+      ;;
+    --quantize_max_percentile)
+      QUANTIZE_MAX_PERCENTILE="$2"
+      shift 2
+      ;;
+    --quantize_mode)
+      QUANTIZE_MODE="$2"
+      shift 2
+      ;;
+    --quantize_input_data)
+      QUANTIZE_INPUT_DATA="$2"
+      shift 2
+      ;;
+    --quantize_output_path)
+      QUANTIZE_OUTPUT_PATH="$2"
+      shift 2
+      ;;
+    --pack_output_path)
+      PACK_OUTPUT_PATH="$2"
+      shift 2
+      ;;
+    --)
+      shift
+      break
+      ;;
+    *)
+      echo "[ERROR] getopt internal error"
+      exit 2
+      ;;
+  esac
+done
+
+# source configuration file
+if [ -e "$CONFIG_FILE" ]; then
+  source "$CONFIG_FILE"
```

Comment:
Executing the config file looks quite risky to me. If the config file does not have the form we want, it can do numerous unexpected behaviors. For example, if config file has content `rm -rf .`, one-build will erase all files in the current directory. Or it can download ransomware from web using `wget`.

This may not be acceptable by our release test team. I think it will be better to use a text format (json or ini) for config file and parse the file to get the options.

Reply:
FYI, here is a simple ini parser written in bash script.  https://gist.github.com/splaspood/1473761